examples/pytorch/mnist-ddp/mnist.py

```diff
@@ -0,0 +1,344 @@
+        # and reduce memory usage
+        with torch.no_grad():
+            # Iterate over mini-batches from the evaluation set
+            for inputs, labels in test_loader:
+                samples += len(inputs)
+                # Copy the data to the GPU device if available
+                inputs, labels = inputs.to(device), labels.to(device)
+                outputs = model(inputs)
+                # Sum up batch loss
+                local_loss += criterion(outputs, labels).item()
+                # Get the index of the max log-probability
+                pred = outputs.argmax(dim=1, keepdim=True)
+                local_correct += pred.eq(labels.view_as(pred)).sum().item()
+
+        local_accuracy = 100.0 * local_correct / samples
+
+        header = f"{'-'*15} Epoch {epoch} Evaluation {'-'*15}"
+        print(f"\n{header}\n")
+        # Log local metrics on each rank
+        print(f"Local rank {rank}:")
+        print(f"- Loss: {local_loss / samples:.4f}")
+        print(f"- Accuracy: {local_correct}/{samples} ({local_accuracy:.0f}%)\n")
+
+        # To Tensors so local metrics can be globally reduced across ranks
+        global_loss = torch.tensor([local_loss], device=device)
+        global_correct = torch.tensor([local_correct], device=device)
+
+        # Reduce the metrics on rank 0
+        dist.reduce(global_loss, dst=0, op=torch.distributed.ReduceOp.SUM)
+        dist.reduce(global_correct, dst=0, op=torch.distributed.ReduceOp.SUM)
+
+        # Log the aggregated metrics only on rank 0
+        if rank == 0:
+            global_loss = global_loss / len(test_loader.dataset)
+            global_accuracy = (global_correct.double() / len(test_loader.dataset)) * 100
+            global_correct = global_correct.int().item()
+            samples = len(test_loader.dataset)
+            print("Global metrics:")
+            print(f"- Loss: {global_loss.item():.6f}")
+            print(
+                f"- Accuracy: {global_correct}/{samples} ({global_accuracy.item():.2f}%)"
+            )
+        else:
+            print("See rank 0 logs for global metrics")
+        print(f"\n{'-'*len(header)}\n")
+
+    dist.init_process_group(backend=backend)
+
+    torch.manual_seed(seed)
+    rank = int(os.environ["RANK"])
+    local_rank = int(os.environ["LOCAL_RANK"])
+
+    model = Net()
+
+    use_cuda = torch.cuda.is_available()
+    if use_cuda:
+        if backend != torch.distributed.Backend.NCCL:
+            print(
+                "Please use NCCL distributed backend for the best performance using NVIDIA GPUs"
+            )
+        device = torch.device(f"cuda:{local_rank}")
+        model = DistributedDataParallel(model.to(device), device_ids=[local_rank])
+    else:
+        device = torch.device("cpu")
+        model = DistributedDataParallel(model.to(device))
+
+    transform = transforms.Compose(
+        [
+            transforms.ToTensor(),
+            transforms.Normalize((0.5,), (0.5,)),
+        ]
+    )
+
```

Comment:
Can we refactor this part to make it as simple as here: https://github.com/kubeflow/training-operator/blob/master/examples/pytorch/image-classification/Train-CNN-with-FashionMNIST.ipynb
I think, we can significantly reduce number of code.